dotCMS/src/main/java/com/dotcms/rest/api/v1/user/UserResource.java

```diff
@@ -339,6 +337,28 @@ public final Response loginAs(@Context final HttpServletRequest request, final L
 		return response;
 	}
 
+	private void setCurrentSite(final HttpServletRequest req, final String userID) throws DotDataException, DotSecurityException {
+		final HttpSession session = req.getSession();
+		final User user = APILocator.getUserAPI().loadUserById(userID);
+		final String currentSiteID = (String) session.getAttribute(com.dotmarketing.util.WebKeys.CMS_SELECTED_HOST_ID);
+		Host currentSite = null;
+
+		try {
+			currentSite = APILocator.getHostAPI().find(currentSiteID, user, false);
+		} catch (DotSecurityException e) {
+			final List<Host> sites = APILocator.getHostAPI().findAll(user, false);
```

Comment:
Pulling all the sites is a very heavy operation and more if we just need at the end one element, the `findAll` method underneath uses the `APILocator.getContentletAPI().search` method which allow the use of limits, one option is to create a new version of the `findAll` method that accepts `limit`, `offset`, `sortBy` and you call using `1` as limit.